java/ql/src/experimental/Security/CWE/CWE-297/IgnoredHostnameVerification.ql

```diff
@@ -0,0 +1,55 @@
+/**
+ * @name Ignored result of hostname verification
+ * @description The method HostnameVerifier.verify() returns a result of hostname verification.
+ *              A caller has to check the result and drop the connection if the verification failed.
+ * @kind problem
+ * @problem.severity error
+ * @precision medium
+ * @id java/ignored-hostname-verification
+ * @tags security
+ *       external/cwe/cwe-295
+ */
+
+import java
+import semmle.code.java.controlflow.Guards
+import semmle.code.java.dataflow.DataFlow
+
+private class HostnameVerificationCall extends MethodAccess {
+  HostnameVerificationCall() {
+    getMethod()
+        .getDeclaringType()
+        .getASupertype*()
+        .hasQualifiedName("javax.net.ssl", "HostnameVerifier") and
+    getMethod().hasStringSignature("verify(String, SSLSession)")
+  }
+
+  predicate ignored() {
```

Comment:
Maybe this predicate should be named `isIgnored()` or `isResultIgnored()` to comply with the [style guide](https://github.com/github/codeql/blob/main/docs/ql-style-guide.md#naming).

Reply:
> Wouldn't it make sense to filter out cases where the "ignored" value is returned from/inside another `HostnameVerifier`? https://lgtm.com/query/1603409360070422690/

@intrigus-lgtm Makes sense to me. I've updated the query to ignore calls wrapped by another hostname verifier. Thank you for the suggestion!